packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -654,6 +665,37 @@ export function createHooksInternal<
     });
     return hook;
   }
+
+  const useQueries: UseQueries<TRouter> = (queriesCallback, context) => {
+    const { ssrState, queryClient, prefetchQuery, client } = useContext();
+
+    const proxy = createUseQueriesProxy(client);
+
+    const queries = queriesCallback(proxy);
+
+    if (typeof window === 'undefined' && ssrState === 'prepass') {
+      for (const query of queries) {
+        const queryOption = query as TRPCQueryOptions<any, any, any, any>;
+        if (
+          queryOption.trpc?.ssr !== false &&
+          !queryClient
+            .getQueryCache()
+            .find(getArrayQueryKey(queryOption.queryKey!, 'query'))
+        ) {
+          void prefetchQuery(queryOption.queryKey as any, queryOption as any);
+        }
+      }
+    }
+
+    return __useQueries({
+      queries: queries.map((query) => ({
+        ...query,
+        queryKey: getArrayQueryKey(query.queryKey, 'query'),
+      })) as readonly UseQueryOptions[],
```

Comment:
I hate this....

Maybe we can make sure that `getArrayQueryKey()` of a result of the same fn is stable, then we can do this further "up" and only once?

Reply:
might be awkward because `prefetchQuery` expects the old style query key